### PR TITLE
    refactor worker_agent and brain because the redisDB data structure changed

### PR DESCRIPTION
**Pull request by uniray7**

### CODE REVIEW

File: framework/jagereye/brain/worker_agent.py [hunk 1]
```diff
@@ -18,88 +18,80 @@ def __init__(self, typename, mem_db):
...
         """
-        anal_worker_id = None
-        if (worker_id and analyzer_id):
-            anal_worker_id = self._get_search_key(analyzer_id=analyzer_id, worker_id=worker_id)
-        elif worker_id:
-            # TODO(Ray): check 'result', error handler
-            result = await self._mem_db.keys(self._get_search_key(worker_id=worker_id))
-            if not result:
+        # set 'workerId' field in analyzer table
+        await self._mem_db.set(self._get_anal_key(anal_id), worker_id)
```
Comment (weitingchou) [inline framework/jagereye/brain/worker_agent.py]: Should use 'MSET' to replace these four set operations into one atomic set operation: https://redis.io/commands/mset

File: framework/jagereye/brain/brain.py [hunk 2]
```diff
@@ -223,20 +223,19 @@ def __init__(self, typename, ch_public='public_brain',
...
-                                format(get_func_name(), worker_obj['status']))
+                              format(get_func_name(), worker_status))
+                        return
+                    # update worker status
+                    await self._worker_agent.update_status(WorkerStatus.HSHAKE_1.name, worker_id=worker_id)
+                    hshake_reply = {
+                        'verb': 'hshake-2',
+                        'context': context
+                    }
+                    await self._nats_cli.publish(ch_brain_to_worker, str(hshake_reply).encode())
```
[the remark this answers is not in the document]
Comment (uniray7) [reply framework/jagereye/brain/brain.py]: https://github.com/SuJiaKuan/jagereye/pull/39/commits/14849998b35604c4f2d0051da4be26e66228d8fa

File: framework/jagereye/brain/worker_agent.py (hunk 1 above)
Comment (uniray7) [reply framework/jagereye/brain/worker_agent.py]: https://github.com/SuJiaKuan/jagereye/pull/39/commits/c1240b18dfda89046d0b2b4787505bde87a95314

File: framework/jagereye/brain/worker_agent.py [hunk 3]
```diff
@@ -62,6 +62,30 @@ def _get_worker_key(self, worker_id, field):
...
+        # mget 'status', 'pipelines'
+        mget_cmds = []
+        mget_cmds.append(self._get_worker_key(worker_id, 'status'))
+        mget_cmds.append(self._get_worker_key(worker_id, 'pipelines'))
+        result = await self._mem_db.mget(*mget_cmds)
+        status = result[0].decode()
+        pipelines = jsonify(result[1].decode())
+        return status, pipelines
+
+
```
Comment (weitingchou) [inline framework/jagereye/brain/worker_agent.py]: Too many blank lines.

**Pull request merged**